Emby.Server.Implementations/HttpServer/HttpListenerHost.cs

```diff
@@ -651,36 +642,41 @@ public Task ProcessWebSocketRequest(HttpContext context)
             return _socketListener.ProcessWebSocketRequest(context);
         }
 
-        // this method was left for compatibility with third party clients
-        private static string NormalizeEmbyRoutePath(string path)
+        private static string NormalizeUrlPrefix(string baseUrl)
         {
-            if (path.StartsWith("/", StringComparison.OrdinalIgnoreCase))
+            if (string.IsNullOrWhiteSpace(baseUrl))
             {
-                return "/emby" + path;
+                // If baseUrl is empty, return an empty prefix string
+                return "";
+            }
+            else
+            {
+                if (baseUrl.StartsWith("/"))
+                {
+                    // If baseUrl was configured with a leading slash, just return it as-is
+                    return baseUrl;
+                }
+                else
+                {
+                    // If baseUrl was not configured with a leading slash, append one for consistency
+                    return "/" + baseUrl;
+                }
             }
-
-            return "emby/" + path;
         }
 
-        // this method was left for compatibility with third party clients
-        private static string NormalizeMediaBrowserRoutePath(string path)
+        private static string NormalizeEmbyRoutePath(string baseUrl, string path)
         {
-            if (path.StartsWith("/", StringComparison.OrdinalIgnoreCase))
-            {
-                return "/mediabrowser" + path;
-            }
+            return NormalizeUrlPrefix(baseUrl) + "/emby" + path;
+        }
 
-            return "mediabrowser/" + path;
+        private static string NormalizeMediaBrowserRoutePath(string baseUrl, string path)
+        {
+            return NormalizeUrlPrefix(baseUrl) + "/mediabrowser" + path;
```

Comment:
I guess this would break if `path` doesn't start with a `/` (same for `NormalizeEmbyRoutePath`)

Reply:
That's why the normalization function makes sure there's always a `/`, either because the user entered it, or it prepends it.

Reply:
Oh I see what you mean, in the second part. I guess I'll have to create a "normalize path" function too to ensure this. What shit code.

Reply:
Function created, does the same thing as the other normalize - check if there's actually a `/` at the start of the path, and prepend one if not.